### PR TITLE
Migrate dashboards to RTK

### LINKED ISSUE #40791: dashboards

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #40791

### How to verify

Things work like they used to.
CI is all green.

### CODE REVIEW

File: frontend/src/metabase/api/tags/utils.ts [hunk 1]
```diff
@@ -158,6 +130,58 @@ export function provideCollectionTags(
...
+export function provideDatabaseTags(
+  database: Database,
+): TagDescription<TagType>[] {
+  return [
+    idTag("database", database.id),
+    ...(database.tables ? provideTableListTags(database.tables) : []),
+  ];
+}
+
+export function provideDashboardListTags(
```
Comment (kamilmielnik) [inline frontend/src/metabase/api/tags/utils.ts:163]: This function and the next one are new.
Other have been moved around to keep alphabetical order.

File: frontend/src/metabase/dashboard/containers/ArchiveDashboardModal.jsx [hunk 2]
```diff
@@ -9,11 +9,11 @@ import _ from "underscore";
 
 import ArchiveModal from "metabase/components/ArchiveModal";
 import Collection from "metabase/entities/collections";
-import Dashboard from "metabase/entities/dashboards";
+import Dashboards from "metabase/entities/dashboards";
```
Comment (kamilmielnik) [inline frontend/src/metabase/dashboard/containers/ArchiveDashboardModal.jsx:12]: Just renaming for consistency

File: frontend/src/metabase/api/dashboard.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  provideDashboardTags,
+} from "./tags";
+
+export const dashboardApi = Api.injectEndpoints({
+  endpoints: builder => ({
+    listDashboards: builder.query<Dashboard[], ListDashboardsRequest | void>({
+      query: () => ({
+        method: "GET",
+        url: "/api/dashboard",
+      }),
```
Comment (nemanjaglumac) [inline frontend/src/metabase/api/dashboard.ts:26]: Where are we using the query function? Should it accept `body`?

```suggestion
    listDashboards: builder.query<Dashboard[], ListDashboardsRequest | void>({
      query: body => ({
        method: "GET",
        url: "/api/dashboard",
        body.
      }),
```

File: frontend/src/metabase/entities/dashboards.js [hunk 4]
```diff
@@ -119,18 +156,13 @@ const Dashboards = createEntity({
   },
 
   reducer: (state = {}, { type, payload, error }) => {
-    if (type === FAVORITE_ACTION && !error) {
-      return assocIn(state, [payload, "favorite"], true);
-    } else if (type === UNFAVORITE_ACTION && !error) {
```
Comment (nemanjaglumac) [inline frontend/src/metabase/entities/dashboards.js:124]: Were these "bookmarks" before it was cool? :D

File: frontend/src/metabase/api/dashboard.ts (hunk 3 above)
Comment (uladzimirdev) [reply frontend/src/metabase/api/dashboard.ts:26]: if it's GET, I think it shouldn't

Comment (kamilmielnik) [reply frontend/src/metabase/api/dashboard.ts]: > Should it accept `body`?

Yes, good catch :+1: 
5b5affa6651b7b9b9ca70beca7fb1bc53a5f193d

> if it's GET, I think it shouldn't

It actually should.

File: frontend/src/metabase/entities/dashboards.js [hunk 5]
```diff
@@ -119,18 +156,13 @@ const Dashboards = createEntity({
...
-      return assocIn(state, [payload, "favorite"], false);
-    } else if (type === COPY_ACTION && !error && state[""]) {
+    if (type === COPY_ACTION && !error && state[""]) {
```
Comment (uladzimirdev) [inline frontend/src/metabase/entities/dashboards.js:159]: I know it's written by someone else, but what's the purpose of `state[""]`?
image

**Comment by uladzimirdev** [comment]
So did I get it right, `favorite` and `unfavorite` were not used?

**Review by uladzimirdev — APPROVED** [review]
Except minor questions, LGTM

File: frontend/src/metabase/entities/dashboards.js (hunk 5 above)
Comment (kamilmielnik) [reply frontend/src/metabase/entities/dashboards.js:159]: No idea. But it's potentially not useless:

```
x = { "": "hello" };
x[""] // "hello"
```

**Comment by kamilmielnik** [comment]
> So did I get it right, `favorite` and `unfavorite` were not used?

Unused and the endpoints actually don't exist in BE.

**Review by nemanjaglumac — APPROVED** [review]

**Pull request merged**